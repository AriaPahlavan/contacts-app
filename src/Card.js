import React from 'react';


const Card = (props) => {
    return (
        <div className="tc dib br3 bw2 pa3 ma2 bg-light-green grow shadow-1">
            <img src="https://robohash.org/test?set=set4" alt="avatar"/>
            <div>
                <h2>Jane Doe</h2>
                <h3>jane.doe@gmail.com </h3>
            </div>
        </div>
    );
};


export default Card;