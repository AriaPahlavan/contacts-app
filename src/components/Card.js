import React from 'react';

const Card = ({id, name, email}) => {
    return (<div className="tc dib br3 bw2 pa3 ma2 bg-light-green grow shadow-1">
        <img src={`https://robohash.org/${name}?set=set4&size=150x150`} alt="avatar"/>
        <div>
            <h2>{name}</h2>
            <h3>{email}</h3>
        </div>
    </div>);
};

export default Card;
