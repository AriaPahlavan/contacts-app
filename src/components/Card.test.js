import {shallow, mount, render} from 'enzyme';
import React from 'react';
import Card from './Card';

it('ensurs Card rendered like snapshot', () => {
  expect(shallow(<Card />)).toMatchSnapshot()
})
