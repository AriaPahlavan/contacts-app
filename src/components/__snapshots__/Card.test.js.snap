// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ensurs Card rendered like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="avatar"
          src="https://robohash.org/undefined?set=set4&size=150x150"
/>,
        <div>
          <h2 />
          <h3 />
</div>,
      ],
      "className": "tc dib br3 bw2 pa3 ma2 bg-light-green grow shadow-1",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "avatar",
          "src": "https://robohash.org/undefined?set=set4&size=150x150",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h2 />,
            <h3 />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h2",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "h3",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="avatar"
            src="https://robohash.org/undefined?set=set4&size=150x150"
/>,
          <div>
            <h2 />
            <h3 />
</div>,
        ],
        "className": "tc dib br3 bw2 pa3 ma2 bg-light-green grow shadow-1",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "avatar",
            "src": "https://robohash.org/undefined?set=set4&size=150x150",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h2 />,
              <h3 />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h2",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h3",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
